contracts/token/PartialCommonOwnership721.sol

```diff
@@ -40,6 +37,53 @@ contract PartialCommonOwnership721 is
     ERC721(name_, symbol_)
   {}
 
+  //////////////////////////////
+  /// Internal Methods
+  //////////////////////////////
+
+  /// @notice Mints a new token.
+  /// @param tokenId_ Token's ID.
+  /// @param leasee_ Token's leasee.
+  /// @param deposit_ Token's deposit.
+  /// @param valuation_ Leasee's self assessed valuation of the token.
+  /// @param beneficiary_ Beneficiary of the token's taxation.
+  /// @param taxRate_ Tax rate (numerator).
+  /// @param collectionFrequency_ Tax collection frequency.
+  function _mint(
+    uint256 tokenId_,
+    address leasee_,
+    uint256 deposit_,
+    uint256 valuation_,
+    address payable beneficiary_,
+    uint256 taxRate_,
+    uint256 collectionFrequency_
+  ) internal {
+    ERC721._safeMint(leasee_, tokenId_);
+    _setDeposit(tokenId_, deposit_);
+    _setValuation(tokenId_, valuation_);
+    _setBeneficiary(tokenId_, beneficiary_);
+    _setTaxRate(tokenId_, taxRate_);
+    _setTaxPeriod(tokenId_, collectionFrequency_);
+  }
+
+  /// @notice Burns a token.
+  /// @param tokenId_ ID of token to burn.
+  function _burn(uint256 tokenId_) internal override _collectTax(tokenId_) {
+    // Return the current owner's deposit.
+    _withdrawDeposit(tokenId_, depositOf(tokenId_));
+
+    // Burn token
+    ERC721._burn(tokenId_);
+
+    // Delete state
+    delete _beneficiaries[tokenId_];
```

Comment:
It's definitely more complicated with the module setup; one of the major changes is that each module manages its own state, which enables modifications to how the state is represented.  E.g. the deposits module & state can be easily swapped for changes like #29 without impacting non-deposit state.  Unfortunately I think this would be harder with structs.  Let's plan on exploring these questions down the road.